src/schema/users.ts

```diff
@@ -963,16 +980,25 @@ export const resolvers: IResolvers<any, Context> = {
       { id, after, before }: ReadingHistyoryArgs,
       ctx: Context,
     ): Promise<GQLReadingRankHistory[]> =>
-      getUserReadHistory({ con: ctx.con, userId: id, after, before }),
-    userStreak: async (_, __, ctx: Context, info): Promise<GQLUserStreak> => {
+      getUserReadHistory({
+        con: ctx.con,
+        userId: id,
+        after: new Date(after),
+        before: new Date(before),
+      }),
+    userStreak: async (
+      _,
+      __,
+      ctx: AuthContext,
```

Comment:
Query `userStreak` has `@auth` directive in gql definition so `ctx.userId` is required to be present, this is why it is using `AuthContext` type.

Reply:
Other queries that do not have `@auth` can just use `Context` as before.